libraries/scripting/dependencies-maven/src/kotlin/script/experimental/dependencies/maven/MavenDependenciesResolver.kt

```diff
@@ -89,21 +90,22 @@ class MavenDependenciesResolver : ExternalDependenciesResolver {
         val url = repositoryCoordinates.toRepositoryUrlOrNull()
             ?: return false.asSuccess()
         val repoId = repositoryCoordinates.string.replace(FORBIDDEN_CHARS, "_")
-        val repo = RemoteRepository.Builder(repoId, "default", url.toString())
-        if (repositoryCoordinates is MavenRepositoryCoordinates) {
-            val username = repositoryCoordinates.username?.let(::tryResolveEnvironmentVariable)
-            val password = repositoryCoordinates.password?.let(::tryResolveEnvironmentVariable)
-            if (username != null) {
-                val auth = AuthenticationBuilder().apply {
-                    addUsername(username)
-                    if (password != null) {
-                        addPassword(password)
+        val repo = RemoteRepository.Builder(repoId, "default", url.toString()).apply {
+            setAuthentication(
+                AuthenticationBuilder().apply {
+                    with(options) {
+                        addUsername(username?.let(::tryResolveEnvironmentVariable))
```

Comment:
I think that implemented approach is more in style of `org.eclipse.aether.repository.Authentication` interface: we simply put into the container all that we have, and consumers in the resolver internals decide what they will use. I think that this logic is more clear and more correct. Empty authorization doesn't differ (in logic) from null one, the cost here is insignificant

Reply:
I do not see any "style" in the `org.eclipse.aether.repository.Authentication` that would suggest such implementation. Can you please show, what you mean? Or maybe it is documented somewhere?
Without some examples or explicit recommendations, I would consider this approach ambiguous, and therefore not desirable.

Reply:
Each Authentication populates AuthenticationContext in the `fill` method. `AuthenticationContext` has all the data in the raw map `org.eclipse.aether.repository.AuthenticationContext#authData`, and then each consumer uses it in its own way. I mean that all the logic about nullability and emptiness is on the lower level, these checks will be performed anyway.
For example, see how authorization is used in wagon transporter: it is first converted to another instance (`org.eclipse.aether.transport.wagon.WagonTransporter#getProxy`) and then these values are used with all the checks performed here: `org.apache.maven.wagon.shared.http.AbstractHttpClientWagon#openConnectionInternal`. I don't think we should care about it: some implementations may use password without username or username without password, it should not be solved here

Reply:
Since it is a place that I stumbled upon, it may look suspicious to others in the future too. I'd suggest at least adding an appropriate comment with the link to implementation (and/or documentation) as an explanation.